include/boost/decimal/decimal32.hpp

```diff
@@ -1197,7 +1202,11 @@ constexpr auto mixed_equality_impl(decimal32 lhs, Integer rhs) noexcept -> bool
         }
     }
 
-    Unsigned_Integer rhs_significand {rhs < 0 ? detail::apply_sign(rhs) : rhs};
+    const auto rhs_significand =
+        static_cast<Unsigned_Integer>
+        (
+            (rhs < 0) ? static_cast<Unsigned_Integer>(detail::apply_sign(rhs)) : static_cast<Unsigned_Integer>(rhs)
+        );
```

Comment:
I added `detail::make_positive_unsigned` to avoid this situation now. All you need is:

`Unsigned_Integer rhs_significand {detail::make_positive_unsigned(res)};`

Reply:
OK.